File: code_samples/scala/design_patterns_adapter.scala
/**
 * Design Patterns: Adapter
 * AI/ML Training Sample
 */

class Adapter {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object AdapterExample {
  def main(args: Array[String]): Unit = {
    val instance = new Adapter()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/csharp/design_patterns_adapter.cs
using System;

/// <summary>
/// Design Patterns: Adapter
/// AI/ML Training Sample
/// </summary>
public class Adapter
{
    private string data;
    
    public Adapter()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new Adapter();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}
